### ISSUE TITLE
How to reconstruct 3D scene from just one rgb image with pretrained model?

### ISSUE THREAD

**Issue by tianyilt** [DLR-RM/SingleViewReconstruction#4]
Hello @themasterlink 
Thanks for your great work for Indoor SingleViewReconstruction. Sorry that I'm newer in this deep learning task. 
I'm confused about how to reconstruct a 3D scene from just one RGB image with a pre-trained model.
The input image may be like the following:
![test](https://user-images.githubusercontent.com/41582525/97385890-a8ab4d00-190d-11eb-8bbc-cc422e2b9b6a.jpg)

After analyzing the _readme_ of all subfolders and discussion in #3, I am still very confused about how a jpg --> 
I try to develop the SingleJPGReconstruction.py with the following step:

1. Gain normal_img from the test.jpg with UNetNormalGen:
I am blocked in h5py input and _setting_file.yml_.
the input of _generate_encoded_outputs.py_ is h5py and settings_file.yml that is seemed to be generated by BlenderProc. I have no idea how to generate this input from a single image without BlenderProc.
I tried to use command `python generate_predicted_normals.py --model_path model/model.ckpt --path ../data` in _SingleViewReconstruction/SingleViewReconstruction_.

1. Combine normal_image and color_image into one hdf5 file.
 #3 says the data should minus normal_mean_image and color_mean_image
The color_normal_mean.hdf5 is used here:
```
SingleViewReconstruction/SingleViewReconstruction/generate_tf_records.py

Line 123 in f1475ca

 normal_o -= normal_mean_img 
and here:

SingleViewReconstruction/SingleViewReconstruction/src/DataSetLoader.py

Line 95 in f1475ca
color_img -= self.mean_img 
```

3. try _predict_datapoint.py_
I tried command `python predict_datapoint.py data/color_normal_mean.hdf5 --output OUTPUT --use_pretrained_weights`.
there wasn't train*.tf_record. 

The plan of course didn't work. Could you give me some help on how to make it? Thanks sincerely.

**Comment by themasterlink** [comment]
Hey,

I will write such a script for you, but this will need some time.

Best,
Max

**Comment by tianyilt** [comment]
Thank you very much！

**Comment by RamyaThanga** [comment]
Is this script available? Or please help us with steps and direction so that we can make it work?

Thank you !

**Comment by tianyilt** [comment]
> Is this script available? Or please help us with steps and direction so that we can make it work?
> 
> Thank you !

I solve the problem by referring to [LLFF ](https://github.com/Fyusion/LLFF#using-your-own-input-images-for-view-synthesis)and [NeRF](https://github.com/bmild/nerf) projects. LLFF is very easy to deploy with external own dataset by docker.

**Comment by themasterlink** [comment]
I have finished the script, I have changed some stuff to make it really simple, it takes care of everything as gives you the result in the end.

It allows to also generate the ground truth data, to check how good the reconstruction is. It works with synthetic normals and the once generated from the UNet.

https://github.com/DLR-RM/SingleViewReconstruction/blob/master/run_on_example_scenes_from_scenenet.py
